guides/source/asset_pipeline.md

```diff
@@ -8,479 +8,616 @@ This guide covers the asset pipeline.
+For example, converting CoffeeScript to JavaScript.
 
 ```javascript
-import React from "react"
+alert "Hello, world!"
 ```
 
-You would have to define it like this to make it work:
+After transpilation, this code becomes:
 
 ```javascript
-import React from "https://ga.jspm.io/npm:react@17.0.2/index.js"
+alert("Hello, world!");
 ```
 
-Here comes the import map, we define the `react` name to be pinned to the `https://ga.jspm.io/npm:react@17.0.2/index.js` address. With such information our browser accepts the simplified `import React from "react"` definition. Think of import map as about an alias for the library source address.
+In the past, pre-processors like [Sass](https://sass-lang.com/) and
+[Less](https://lesscss.org/) were essential for CSS features such as variables
+and nesting. Today, modern CSS supports these natively, reducing the need for
+transpilation.
 
-### Usage
+### Bundling ❌
 
-With `importmap-rails` you create the importmap configuration file pinning the library path to a name:
+Bundling combines multiple files into one to reduce the number of HTTP requests
+a browser needs to make to render a page.
 
-```ruby
-# config/importmap.rb
-pin "application"
-pin "react", to: "https://ga.jspm.io/npm:react@17.0.2/index.js"
-```
-
-All of the configured import maps should be attached in `<head>` element of your application by adding  `<%= javascript_importmap_tags %>` . The `javascript_importmap_tags` renders a bunch of scripts in the `head` element:
+For example, if your application has three JavaScript files:
+- menu.js
+- cart.js
+- checkout.js
 
-- JSON with all configured import maps:
+Bundling will merge these into a single application.js file.
 
-```html
-<script type="importmap">
-{
-  "imports": {
-    "application": "/assets/application-39f16dc3f3....js"
-    "react": "https://ga.jspm.io/npm:react@17.0.2/index.js"
-  }
-}
-</script>
-```
-
-- Entrypoint for loading JavaScript from `app/javascript/application.js`:
-
-```html
-<script type="module">import "application"</script>
+```javascript
+// Contents of menu.js, cart.js, and checkout.js are combined here
 ```
 
-NOTE: Before v2.0.0, `importmap-rails` put [`Es-module-shims`](https://github.com/guybedford/es-module-shims) in the output of `javascript_importmap_tags` as a polyfill to ensure support for import maps on older browsers. However, with the native support for import maps in all major browsers, v2.0.0 has dropped the bundled shim. If you want to support legacy browsers that lack support for import maps, manually insert `Es-module-shims` before `javascript_importmap_tags`. For more information, refer to [README for importmap-rails](https://github.com/rails/importmap-rails?tab=readme-ov-file#supporting-legacy-browsers-such-as-safari-on-ios-15).
+This was crucial with HTTP/1.1, which limited 6-8 simultaneous connections per
+domain. With HTTP/2, browsers fetch multiple files in parallel, making bundling
+less critical for modern applications.
```

Comment:
```suggestion
domain. With HTTP/2, browsers can fetch multiple files in parallel, making bundling
less critical for modern applications.
```

Reply:
Using "can" in the suggestion implies optionality. However, I'm under the impression that HTTP/2 fetches multiple files in parallel with certainty.